### PR TITLE
acr cmk enabled

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'ACR CMK Encryption Enabled',
+    category: 'Container Registry',
+    domain: 'Containers',
+    description: 'Ensure that Microsoft Azure Container registries have CMK Encryption Enabled.',
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:8]:
```suggestion
    description: 'Ensure that Microsoft Azure Container registries have CMK encryption enabled.',
```

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            if (!containerRegistries.data.length) {
+                helpers.addResult(results, 0, 'No existing Container registries found', location);
+                return rcb();
+            } 
+            
+            for (let registry of containerRegistries.data) {
+                if (!registry.id) continue;
+
+                if (registry.encryption && registry.encryption.status && registry.encryption.status.toLowerCase() === 'enabled') {
+                    helpers.addResult(results, 0, 'Container Registry have cmk encryption enabled', location, registry.id);
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:39]:
```suggestion
                    helpers.addResult(results, 0, 'Container Registry has CMK encryption enabled', location, registry.id);
```

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                } else {
+                    helpers.addResult(results, 2, 'Container Registry does not have cmk encryption enabled', location, registry.id);
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:41]:
```suggestion
                    helpers.addResult(results, 2, 'Container Registry does not have CMK encryption enabled', location, registry.id);
```

File: plugins/azure/containerregistry/acrCMKEncryption.js (hunk 1 above)
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:8]:
```suggestion
    description: 'Ensure that Microsoft Azure Container registries have CMK encryption enabled.',
```

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'ACR CMK Encryption Enabled',
+    category: 'Container Registry',
+    domain: 'Containers',
+    description: 'Ensure that Microsoft Azure Container registries have CMK Encryption Enabled.',
+    more_info: 'A customer-managed key gives you the ownership to bring your own key in Azure Key Vault. When you enable a customer-managed key, you can manage its rotations, control the access and permissions to use it, and audit its use.',
+    recommended_action: 'Modify Container registries and enable encryption using cmk.',
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:10]: Can we modify the encryption of container registry once its created?

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+const async = require('async');
+const helpers = require('../../../helpers/azure');
+
+module.exports = {
+    title: 'ACR CMK Encryption Enabled',
```
Comment (alphadev4) [inline plugins/azure/containerregistry/acrCMKEncryption.js:5]:
```suggestion
    title: 'ACR CMK Encryption',
```

File: plugins/azure/containerregistry/acrCMKEncryption.js (hunk 4 above)
Comment (abdullahaslam306) [reply plugins/azure/containerregistry/acrCMKEncryption.js]: This one is resolved. No we cannot change encryption settings once it is created

File: plugins/azure/containerregistry/acrCMKEncryption.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+            if (!containerRegistries.data.length) {
+                helpers.addResult(results, 0, 'No existing Container registries found', location);
+                return rcb();
+            } 
+            
+            for (let registry of containerRegistries.data) {
+                if (!registry.id) continue;
+
+                if (registry.encryption && registry.encryption.status && registry.encryption.status.toLowerCase() === 'enabled') {
```
Comment (AkhtarAmir) [inline plugins/azure/containerregistry/acrCMKEncryption.js:38]: Does this mean it is CMK encrypted? Or just encrypted

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/azure/containerregistry/acrCMKEncryption.js (hunk 6 above)
Comment (abdullahaslam306) [reply plugins/azure/containerregistry/acrCMKEncryption.js:38]: registries are encrypted by default and for CMK we can check using these attributes

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**